explorer/app/hooks/useFetchEntities.ts

```diff
@@ -16,14 +17,35 @@ export interface PaginationConfig {
   currentPage: number;
 }
 
+type SortOrderType = "asc" | "desc";
+
+export interface SortConfig {
+  sort: (key?: string, sortOrder?: SortOrderType) => void;
+  sortKey?: string;
+  sortOrder?: SortOrderType;
+}
+
 interface UseEntityListResponse {
   response?: ListResponseType;
   isLoading: boolean;
   pagination?: PaginationConfig;
+  sort?: SortConfig;
 }
 
 const DEFAULT_CURRENT_PAGE = 1;
 
+/**
+ * Retrieves the column set with default true
+ * @param entity current entity config with all columns
+ * @returns string with the default sorted key or the first one
+ */
+const getDefaultSort = (entity?: EntityConfig) => {
+  return (
+    entity?.list?.columns.find((column) => column.sort?.default)?.sort
+      ?.sortKey ?? entity?.list?.columns[0].sort?.sortKey
```

Comment:
Can you also update `list` on `EntityConfig` to be required? I think `detail` can remain as optional but we will always need a list config for an entity. You can then also remove the optional chaining here on `list`.